### PR TITLE
feature: S3C-4500 Add probe server for queue processor

### PR DESCRIPTION
**Pull request by miniscruff**
Add probe server to queue processor with checks for producer and consumer

### CODE REVIEW

File: extensions/replication/queueProcessor/QueueProcessor.js [hunk 1]
```diff
@@ -385,6 +385,45 @@ class QueueProcessor extends EventEmitter {
...
+                component: 'Replication Status Producer',
+                status: 'not ready',
+                site: this.site,
+            });
+        }
+
+        log.debug('verbose liveness', verboseLiveness);
+
+        if (responses.length > 0) {
+            return JSON.stringify(responses);
```
Comment (rahulreddy) [inline extensions/replication/queueProcessor/QueueProcessor.js:441]: If you return here then no response will be send to the client, right?

File: extensions/replication/queueProcessor/task.js [hunk 2]
```diff
@@ -29,16 +32,56 @@ werelogs.configure({ level: config.log.logLevel,
...
+            DEFAULT_LIVE_ROUTE,
+            (res, log) => queueProcessor.handleLiveness(res, log)
+        );
+        probeServer._cbOnListening = done;
+        probeServer.start();
+        return undefined;
+    },
+    done => {
+        metricsProducer.setupProducer(err => {
+            log.error('error starting metrics producer for queue processor', {
```
Comment (rahulreddy) [inline extensions/replication/queueProcessor/task.js]: You need a condition here so that it does not log when there is no error

**Review by rahulreddy — CHANGES_REQUESTED** [review]

File: extensions/replication/queueProcessor/QueueProcessor.js (hunk 1 above)
Comment (miniscruff) [reply extensions/replication/queueProcessor/QueueProcessor.js:441]: The probe server handles sending back bad responses from the string return value. https://github.com/scality/Arsenal/blob/e1a3b05330598a656806e888a760b65711fbfd76/lib/network/probe/ProbeServer.js#L94-L100

**Comment by naren-rajendran** [comment]
Looks good to me.

**Review by rahulreddy — COMMENTED** [review]
LGTM

**Review by naren-rajendran — APPROVED** [review]

File: lib/util/probeUtils.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
```
Comment (naren-rajendran) [inline lib/util/probeUtils.js]: May be these should be constants like in other components! I see that we dont have that in backbeat 7.x!

Comment (miniscruff) [reply lib/util/probeUtils.js]: I could move these into the constants.js if you think that is the better approach, personally not a fan of a 'constants' file as they can get quite a bit and messy over time ( reminds me of static "manager" classes from games... ).

Comment (naren-rajendran) [reply lib/util/probeUtils.js]: Yes I'd prefer constants, seems consistent. Something about calling this a util seems a bit wrong.

Comment (miniscruff) [reply lib/util/probeUtils.js]: yea that is fair, I sort of envisioned it having more stuff in it... but then it didn't. I will move it to the existing constants then.

File: tests/functional/replication/probe.spec.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        // return sample error for liveness
+        mockQp.handleLiveness.returns('error msg');
+        const config = {
+            bindAddress: 'localhost',
+            port: 52555,
+        };
+        startProbeServer(mockQp, config, probeServer => {
+            probeServer.onStop(done);
+            http.get(`http://localhost:52555${DEFAULT_LIVE_ROUTE}`, res => {
+                assert.strictEqual(500, res.statusCode);
```
Comment (rahulreddy) [inline tests/functional/replication/probe.spec.js]: This should be `actual, expected` for the assertion check, I think its the other way around here.

Comment (rahulreddy) [reply tests/functional/replication/probe.spec.js]: Is there any reason its 500 and not 200?

Comment (miniscruff) [reply tests/functional/replication/probe.spec.js]: The liveness check returns an error so the status code is 500, the test confirms the server is up and running and correctly handles the liveness response. The handle liveness is more thoroughly tested in unit tests.

**Review by rahulreddy — APPROVED** [review]

**Pull request merged**